src/messages/MOSDRepOp.h

```diff
@@ -188,6 +206,10 @@ class MOSDRepOp final : public MOSDFastDispatchOp {
     }
     out << ")";
   }
+
+  void clear_buffers() override {
```

Comment:
Reviewer feedback requested:

I’ve moved an ObjectStore::Transaction into the MOSDRepOp message so that it can be encoded when the message is sent rather than when the message is built. I added a clear_buffers() method (copying what is done in MOSDECSubOpWrite) to clean up the transaction and avoid buffers being held referenced. Is this correct?